### PR TITLE
[SDK-702] device id type function and tests

### PR DESCRIPTION
**Pull request by turtledreams**
On going, not finished.

### CODE REVIEW

File: cypress/integration/device_id.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+function initMain(ob) {
+    Countly.init(ob);
+}
+function idType() {
+    return Countly.check_device_id_type();
+}
+
+/**
+device ID types:
```
Comment (ArtursKadikis) [inline cypress/integration/device_id.js:14]: move these

File: cypress/integration/device_id.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
+/* eslint-disable cypress/no-unnecessary-waiting */
+/* eslint-disable require-jsdoc */
+var Countly = require("../../lib/countly");
+var hp = require("../support/helper");
+
+function initMain(ob) {
+    Countly.init(ob);
+}
+function idType() {
```
Comment (ArtursKadikis) [inline cypress/integration/device_id.js:9]: probably not needed, just use the call, Luke!

File: cypress/support/helper.js [hunk 3]
```diff
@@ -48,6 +48,38 @@ var waitFunction = function(startTime, waitTime, waitIncrement, continueCallback
     }
 };
 
+/**
+ *  returns a customized init object
+ * @param {Array} specs - an array of strings that corresponds to desired object customization values. Possible values: 
+ * tests, id, offline, debug
+ * @returns {Object} - final customized object 
+ */
+function giveInitObj(specs) {
```
Comment (ArtursKadikis) [inline cypress/support/helper.js:57]: kill this abomination

File: lib/countly.js [hunk 4]
```diff
@@ -527,6 +555,18 @@
             return false;
         };
 
+        /**
+        * Check and return the current device id type
+        * @returns {number} a number that indicates the device id type
+        */
+        this.check_device_id_type = function() {
```
Comment (ArtursKadikis) [inline lib/countly.js:562]: it should be "get_device_id_type"

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
Device ID type call should return enum. There should be enum values to compare to. Write doc entry.

**Comment by turtledreams** [comment]
Tests are modified. Also currently it seems to be the order of deviceID setting power is like this:
 URL>OFFLINE>USERSET>AUTOGEN. 
Will check our chart and will comment here again if it is proper.
To be continued...

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
Add 8 tests in 2 pairs of 4, and then change device ID or enter offline mode and then change id

**Review by ArtursKadikis — APPROVED** [review]

**Pull request merged**